src/ffi/extra/module.cc

```diff
@@ -30,6 +31,32 @@
 namespace tvm {
 namespace ffi {
 
+/*!
+ * \brief Global modules, i.e. modules that are owned by the runtime and should not be unloaded.
+ * On the frontend, a module is added to the registry if `keep_alive=True` when `load_module` is
+ * called.
+ */
+struct ModuleGlobals {
+  void Add(const Module& m) {
+    std::scoped_lock<std::mutex> lock(mutex);
+    modules.insert(m);
+  }
+
+  void Remove(const Module& m) {
+    std::scoped_lock<std::mutex> lock(mutex);
+    modules.erase(m);
+  }
+
+  static ModuleGlobals* Get() {
+    static ModuleGlobals instance;
+    return &instance;
+  }
+
+ private:
+  std::unordered_set<Module, ObjectPtrHash, ObjectPtrEqual> modules;
```

Comment:
Google c style `modules_`, we can actually consider use Map<Module, int>, which might help reduce ABI bloat as we don't need to have `unordered_set`

Reply:
updated